tests/test_options.py

```diff
@@ -700,16 +700,32 @@ def test_show_default_boolean_flag_name(runner, default, expect):
     assert f"[default: {expect}]" in message
 
 
-def test_show_default_boolean_flag_value(runner):
-    """When a boolean flag only has one opt, it will show the default
-    value, not the opt name.
+def test_show_true_default_boolean_flag_value(runner):
+    """When a boolean flag only has one opt and its default is True,
+    it will show the default value, not the opt name.
+    """
+    opt = click.Option(
+        ("--cache",),
+        is_flag=True,
+        show_default=True,
+        default=True,
+        help="Enable the cache.",
+    )
+    ctx = click.Context(click.Command("test"))
+    message = opt.get_help_record(ctx)[1]
+    assert "[default: True]" in message
+
+
+def test_hide_false_default_boolean_flag_value(runner):
+    """When a boolean flag only has one opt and its default is False,
+    it will not show the default
     """
     opt = click.Option(
         ("--cache",), is_flag=True, show_default=True, help="Enable the cache."
     )
     ctx = click.Context(click.Command("test"))
     message = opt.get_help_record(ctx)[1]
-    assert "[default: False]" in message
+    assert "[default: False]" not in message
```

Comment:
If it tries to show some other default somehow, this test will still pass. Probably `'default' not in message` is safer.

Reply:
I'm concerned the word "default" might feature in the actual help text. Perhaps I could use 
```assert "[default: " not in message```